### ISSUE TITLE
Wrong returncode when using codeclimate as the output format

### ISSUE THREAD

**Issue by SaipranavK** [hadolint/hadolint#688]
- [x] This is a bug report
- [ ] This is a feature request
- [x] I searched existing issues before opening this one

### Expected behavior
The output from the scan should result in returncode 1.

### Actual behavior
The output from the scan results in returncode 0

### Steps to reproduce the behavior
I have a Dockerfile which violates rules with severity above the default failure-threshold info. When I run the scan on this Dockerfile with the output format as codeclimate, it returns 0. The same is not true for other output formats. 

**Output of `hadolint Dockerfile` **
```bash
Dockerfile:6 DL3042 warning: Avoid use of cache directory with pip. Use `pip install --no-cache-dir <package>`
```
**Output of `echo $?` **
```bash
1
```

**Output of `hadolint -f json Dockerfile` **
```bash
[{"file":"Dockerfile","column":1,"message":"Avoid use of cache directory with pip. Use `pip install --no-cache-dir <package>`","code":"DL3042","level":"warning","line":6}]
```
**Output of `echo $?` **
```bash
1
```

**Output of `hadolint -f codeclimate Dockerfile` **
```bash
{"severity":"major","check_name":"DL3042","description":"Avoid use of cache directory with pip. Use `pip install --no-cache-dir <package>`","categories":["Bug Risk"],"type":"issue","location":{"lines":{"end":6,"begin":6},"path":"Dockerfile"}}
```
**Output of `echo $?` **
```bash
0
```

**Dockerfile** (if relevant)
```Dockerfile
FROM python:3.6
ENV PYTHONUNBUFFERED=1
RUN mkdir /mynk
WORKDIR /mynk
COPY requirements.txt /mynk/
RUN pip install -r requirements.txt
COPY . /mynk/
EXPOSE 8000
CMD ["python", "manage.py", "runserver", "0.0.0.0:8000"]
```

**Additional environment details (OS, stack version, etc.)**
OS - Linux

**Comment by vidarlx** [comment]
Have the same, both `info` and `warning` issues result in `1` exit code (DL3047 and DL3018  violations, to be precise)

**Comment by m-ildefons** [comment]
I've followed the commit history back to when that behavior was introduced  in 69bcd34ac6015ef92ee542652b72e76bc495de89. It was never explained in the commit logs why the exit code is forced to `exitSuccess`, but it was there from the beginning. I suspect that the codacy formatter has the same behavior, because when it was introduced the code was inspired by the codeclimate formatter. @lorenzo , if you could share some insight into why these two formatters are forced to `exitSuccess`?

I don't know enough about these two integrations, but I can imagine that exiting with anything other than `exitSuccess` is considered a crash by codeclimate resp. codacy and only the JSON results actually matter anyways.
I believe we can get rid of these special cases and rely on the `--no-fail` option in case codeclimate or codacy expect successful exit codes instead.

**Comment by lorenzo** [comment]
It’s just how codeclimate expects results. If you want the codeclimate format, but with an exit code for failure, try the codeclimate_json formatter.

You dan follow the codeclimate discussion here https://github.com/codeclimate/codeclimate/issues/422
